File: lab_2/filereading.cpp
#include "filereading.h"
#include <fstream>
#include <string>
using namespace std;


string read_file(const char* path) {
    ifstream file(path);
    if (!file.is_open()) {
        throw std::invalid_argument("Uncorrect path or file!");
    }
    string res;
    getline(file, res);
    return res;
}
std::vector<int> read_vector(const string& line) {
    std::vector<int> result;
    for (auto u : line) {
        if (u == '0') {
            result.push_back(0);
        }
        else if (u == '1') {
            result.push_back(1);
        }
        else {
            throw std::runtime_error("Uncorrect file format!");
        }
    }
    return result;
}

std::vector<int> read_sequence(const char* path) {
    return read_vector(read_file(path));
}

File: lab_2/filereading.py
def read_file(path: str) -> str:
    '''
    Чтение строки из файла
    '''
    with open(path, 'r+') as file:
        return file.readline()[:-1]


def read_vector(line: str) -> list[bool]:
    '''
    Чтение вектора из строки
    '''
    result = []
    for symb in line:
        if symb == '0':
            result += [False]
        elif symb == '1':
            result += [True]
        else:
            raise ValueError(
                'Uncorrect file format! Symbol: ' + symb + 'is used!')
    return result


def read_sequence(path: str) -> list[int]:
    '''
    Чтение вектора из файла
    '''
    return read_vector(read_file(path))
